state/block.go

```diff
@@ -0,0 +1,97 @@
+package state
+
+import (
+	"encoding/json"
+	"math/big"
+
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/core/blocktree"
+	"github.com/ChainSafe/gossamer/core/rawdb"
+	"github.com/ChainSafe/gossamer/core/types"
+	"github.com/ChainSafe/gossamer/polkadb"
+)
+
+type blockState struct {
+	bt *blocktree.BlockTree
+	db polkadb.BlockDB
+}
+
+func newBlockState() *blockState {
+	return &blockState{
+		bt: &blocktree.BlockTree{},
+		db: polkadb.BlockDB{},
+	}
+}
+
+// check checks to see if there an error if so writes err + message to terminal
+func check(e error) {
+	if e != nil {
+		panic(e)
+	}
+}
+
+var (
+	// Data prefixes
+	headerPrefix    = []byte("hdr") // headerPrefix + hash -> header
+	blockDataPrefix = []byte("hsh") // blockDataPrefix + hash -> blockData
+)
+
+// headerKey = headerPrefix + hash
+func headerKey(hash common.Hash) []byte {
+	return append(headerPrefix, hash.ToBytes()...)
+}
+
+// blockDataKey = blockDataPrefix + hash
+func blockDataKey(hash common.Hash) []byte {
+	return append(blockDataPrefix, hash.ToBytes()...)
+}
+
+func (bs *blockState) GetHeader(hash common.Hash) types.BlockHeader {
+	var result types.BlockHeader
+
+	data, err := bs.bt.Db.Db.Get(headerKey(hash))
+	check(err)
+
+	err = json.Unmarshal(data, &result)
+	check(err)
+
+	return result
```

Comment:
This isn't a great approach. What happens if we lookup something that doesn't exist? It would be okay to change this to return an error.